python/ray/serve/api.py

```diff
@@ -571,3 +572,100 @@ def delete(name: str, _blocking: bool = True):
     """
     client = get_global_client()
     client.delete_apps([name], blocking=_blocking)
+
+
+@PublicAPI(stability="alpha")
+def multiplexed(
+    func: Optional[Callable[..., Any]] = None, max_num_models_per_replica: int = -1
+):
+    """[EXPERIMENTAL] Defines a function or method used to load multiplexed
+    models in a replica.
+
+    The function can be standalone function or a method of a class. The
+    function must have exactly one argument, the model id of type `str` for the
+    model to be loaded.
+
+    User have to define the function with `async def` and the function must be
+    an async function. It is recommended to define coroutines for long running
+    IO tasks in the function to avoid blocking the event loop.
+
+    The multiplexed function is called to load a model with the given model ID when
+    necessary.
+
+    When the number of models in one replica is larger than max_num_models_per_replica,
+    the models will be unloaded using an LRU policy.
+
+    If you want to release resources after the model is loaded, you can define
+    a `__del__` method in your model class. The `__del__` method will be called when
+    the model is unloaded.
+
+    Example:
+
+    .. code-block:: python
+            from ray import serve
+
+            @serve.deployment
+            class MultiplexedDeployment:
+
+                def __init__(self):
+                    # Define s3 base path to load models
```

Comment:
```suggestion
                    # Define s3 base path to load models.
```
User-facing, use proper grammar/punctuation :)